internal/cmd/egctl/config.go

```diff
@@ -0,0 +1,158 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package egctl
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/yaml"
+
+	"github.com/envoyproxy/gateway/internal/cmd/options"
+	kube "github.com/envoyproxy/gateway/internal/kubernetes"
+)
+
+var (
+	output        string
+	podName       string
+	podNamespace  string
+	containerName string
+	adminPort     int32
+)
+
+func NewConfigCommand() *cobra.Command {
+	cfgCommand := &cobra.Command{
+		Use:     "config",
+		Aliases: []string{"c"},
+		Short:   "Retrieve proxy configuration.",
+		Long:    "Retrieve information about proxy configuration from envoy proxy and gateway.",
+	}
+
+	cfgCommand.AddCommand(proxyCommand())
+
+	flags := cfgCommand.Flags()
+	options.AddKubeConfigFlags(flags)
+
+	cfgCommand.PersistentFlags().StringVarP(&output, "output", "o", "json", "One of 'yaml' or 'json'")
+	cfgCommand.PersistentFlags().StringVar(&containerName, "container-name", "envoy", "Name of container where envoy proxy is installed.")
```

Comment:
the original design doc had `instance-name`, why is `container-name` needed, can we do without adding this arg ?